content/scalability/docs/cluster-services.md

```diff
@@ -0,0 +1,60 @@
+
+Cluster services are expected to have a high up-time and are often critical during outages and for troubleshooting. They should run on dedicated compute instances such as a separate node group or AWS Fargate. This will ensure that the cluster services are not impacted on shared instances by workloads that may be scaling up or using more resources.
+
+## Scale CoreDNS
+
+Scaling CoreDNS has two primary mechanisms. Reducing the number of calls to the CoreDNS service and increasing the number of replicas.
+
+### Reduce external queries by lowering ndots
+
+The ndots setting specifies how many periods in a domain name are considered enough to avoid querying DNS. If your application has an ndots setting of 5 (default) and you request resources from api.example.com (http://api.example.com/) (2 dots) then CoreDNS will be queried for each search domain defined in /etc/resolv.conf for a more specific domain (e.g. api.example.com.namespace.svc.) until all searches are exhausted or a match is found.
+
+You can reduce the number of requests to CoreDNS by lowering the ndots option (https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config) of your workload or fully qualifying your domain requests by including a trailing . (e.g. api.example.com.). If your workload connects to external services via DNS we recommend setting ndots to 2 so workloads do not make unnecessary, cluster DNS queries inside the cluster. You can set a different DNS server and search domain if the workload doesn’t require access to services inside the cluster.
+
+```
+spec:
+  dnsPolicy: "None"
+  dnsConfig:
+    options:
+      - name: ndots
+        value: "2"
+      - name: edns0
+```
+
+If you lower ndots to a value that is too low or the domains you are connecting to do not include enough specificity (including trailing .) then it is possible DNS lookups will fail. Make sure you test how this setting will impact your workloads.
+
+### Scale CoreDNS Horizontally 
+
+CoreDNS instances can scale using a horizontal pod autoscaler rule or you can use [NodeLocal DNS](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/). Scaling replicas horizontally is a good option to avoid failed DNS requests. NodeLocal DNS will require more compute resources in the cluster, but it will avoid failed DNS requests and increase the response time for DNS queries in the cluster.
+
+You can also use the [horizontal proportional scaler](https://github.com/kubernetes-sigs/cluster-proportional-autoscaler) to scale CoreDNS based on the number of nodes or cores in the cluster. This isn’t a direct correlation to request queries, but can be useful depending on your workloads. 
```

Comment:
I think there is guidance in the Kubernetes documentation or the CoreDNS documentation about the recommended ratio of nodes to CoreDNS pods. Can we include that here?